demos/2019_07_Hackfest/Setup.md

```diff
@@ -0,0 +1,157 @@
+    * Run the commands:
+```bash
+$ oc sa get-token -n mig mig
+eyJhbGciOifsfsds8ahmtpZCI6IiJ9fdsfdseyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJtaWciLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoibWlnLXRva2VuLTdxMnhjIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6Im1pZyIsImt1YmVybmss7gc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6IjQ5NjYyZjgxLWEzNDItMTFlOS05NGRjLTA2MDlkNjY4OTQyMCIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDptaWc6bWlnIn0.Qhcv0cwP539nSxbhIHFNHen0PNXSfLgBiDMFqt6BvHZBLET_UK0FgwyDxnRYRnDAHdxAGHN3dHxVtwhu-idHKI-mKc7KnyNXDfWe5O0c1xWv63BbEvyXnTNvpJuW1ChUGCY04DBb6iuSVcUMi04Jy_sVez00FCQ56xMSFzy5nLW5QpLFiFOTj2k_4Krcjhs8dgf02dgfkkshshjfgfsdfdsfdsa8fdsgdsfd8fasfdaTScsu4lEDSbMY25rbpr-XqhGcGKwnU58qlmtJcBNT3uffKuxAdgbqa-4zt9cLFeyayTKmelc1MLswlOvu3vvJ2soFx9VzWdPbGRMsjZWWLvJ246oyzwykYlBunYJbX3D_uPfyqoKfzA
+
+# We need to save the output of the 'get-token', that is the long string we will enter into the mig-ui when we create a new cluster entry.
+```
+
+### Destination Cluster
+1. Provision 4.1 environment
+    * `$ cd 4.x/`
+    * `$ ./create_ocp4_workshop.sh`
+    
+1. Set KUBECONFIG to point to 4.1 environment (Assumes you didn't change `output_dir` var in `mig-agnosticd`)
+```bash
+$ export KUBECONFIG=~/.agnosticd/dymurray-ocp4-cluster/ocp4-workshop_dymurray-ocp4_kubeconfig
+```
+
+1. Install migration workload
+    * `$ cd ../workloads/`
+    * `$ ./deploy_workload.sh -a create migration -v 4`
+
+1. Verify that you see the below pods now running in `mig` namespace, we are looking for pods of `controller-manager`, `velero`, and `migration-ui`
+```bash
+$ oc get pods -n mig
+NAME                                 READY   STATUS      RESTARTS   AGE
+migration-operator-437w4jd-88sf      1/1     Running     0          2m
+controller-manager-79bf7cd7d-99sbr   1/1     Running     0          2m
+migration-ui-6d84bb9879-w52qx        1/1     Running     0          2m
+restic-6lhnp                         1/1     Running     0          2m
+restic-c72gl                         1/1     Running     0          2m
+restic-xf4z2                         1/1     Running     0          2m
+velero-6bf58f4f88-6cpw8              1/1     Running     0          2m
+```
+
+1. Determine the route of `mig-ui`, we will use this later to enable a CORS header on the OCP 3.x side.
+```bash
+$ oc get routes migration -n mig -o jsonpath='{.spec.host}'
+migration-mig.apps.cluster-dymurray-ocp4.dymurray-ocp4.mg.example.com
+```
+
+### Create MSSQL Security Context Constraint
+
+1. Run the following to recreate MSSQL's `scc` on the destination cluster:
+```bash
+$ oc create -f files/mssql-scc.yaml
+```
+
+### Enable CORS Headers on Source cluster
+
+1. Run the following playbook to enable the proper CORS headers on the source cluster
+```bash
+$ cd ../cors/
+$ ansible-playbook cors.yaml
+```
+
+## Prepare to use mig-ui from OCP 4.1 Cluster in your Browser
```

Comment:
I might have missed this, but we should call out need to accept the cert for the source cluster as well.
When we go into the Migration Plan wizard and do namespace discovery, browser will error out if we haven't accepted the source clusters cert prior.

Having user visit the console on source side and accept the cert I think is sufficient.